### PR TITLE
Csse layout 536 input_data

### PR DESCRIPTION
**Pull request by loriab**
~currently atop #357~ 

## Description

* see changelog
* `AtomicResult(AtomicInput)` --> `AtomicResult()` with `AtomicResult.input_data = AtomicInput`
* counterpart to https://github.com/MolSSI/QCEngine/pull/459

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: qcelemental/models/v2/procedures.py [hunk 1]
```diff
@@ -348,11 +354,17 @@ def convert_v(
         if check_convertible_version(version, error="TorsionDriveResult") == "self":
             return self
 
+        opthist_class = next(iter(self.optimization_history.values()))[0].__class__
```
Comment (davidbrownell) [inline qcelemental/models/v2/procedures.py:357]: Nit: Can the `opthist_class` variable be moved into the `if`-clause given it seems to only be used there?

File: qcelemental/tests/test_model_results.py [hunk 2]
```diff
@@ -1062,6 +1104,64 @@ def test_model_survey_dictable(smodel1, smodel2, every_model_fixture, request, s
     assert instance
 
 
+@pytest.mark.parametrize("smodel1,smodel2", _model_classes_struct)
+def test_model_survey_convertable(smodel1, smodel2, every_model_fixture, request, schema_versions):
+    anskey = request.node.callspec.id.replace("None", "v1")
+    # fmt: off
+    ans = [
```
Comment (davidbrownell) [inline qcelemental/tests/test_model_results.py:1111]: Nit: Consider making this a set given that it seems to only be used for lookup and is not used with indexes.

**Review by davidbrownell — APPROVED** [review]

File: qcelemental/models/v2/procedures.py [hunk 3]
```diff
@@ -175,9 +175,12 @@ def convert_v(
         if check_convertible_version(version, error="OptimizationResult") == "self":
             return self
 
+        trajectory_class = self.trajectory[0].__class__
         dself = self.model_dump()
         if version == 1:
+            dself["trajectory"] = [trajectory_class(**atres).convert_v(version) for atres in dself["trajectory"]]
```
Comment (loriab) [inline qcelemental/models/v2/procedures.py:181]:
```suggestion
        dself = self.model_dump()
        if version == 1:
            trajectory_class = self.trajectory[0].__class__
            
            dself["trajectory"] = [trajectory_class(**atres).convert_v(version) for atres in dself["trajectory"]]
```

File: qcelemental/tests/test_model_results.py [hunk 4]
```diff
@@ -1062,6 +1104,64 @@ def test_model_survey_dictable(smodel1, smodel2, every_model_fixture, request, s
...
+        # "v1-TDPtcl"   ,  "v2-TDPtcl"  , 
+        "v1-TDRes"    ,  "v2-TDRes"   , 
+        # "v1-TDProp"   ,  "v2-TDProp"  , 
+        # "v1-MBIn"     ,  "v2-MBIn"    , 
+        # "v1-MBSpec"   ,  "v2-MBSpec"  , 
+        # "v1-MBKw"     ,  "v2-MBKw"    , 
+        # "v1-MBPtcl"   ,  "v2-MBPtcl"  , 
+        # "v1-MBRes"    .  "v2-MBRes"   , 
+        # "v1-MBProp"   ,  "v2-MBProp"  , 
+    ]
```
Comment (loriab) [inline qcelemental/tests/test_model_results.py:1139]:
```suggestion
    }
```

**Review by davidbrownell — APPROVED** [review]

**Pull request merged**